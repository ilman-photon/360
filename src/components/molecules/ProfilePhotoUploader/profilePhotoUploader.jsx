import FileUploadOutlinedIcon from "@mui/icons-material/FileUploadOutlined";
import { Avatar, Box, Button, Stack, Typography } from "@mui/material";
import Image from "next/image";
import { useRef, useState } from "react";
import { colors } from "../../../styles/theme";
import { stringAvatar } from "../../../utils/avatar";

export const ProfilePhotoUploader = ({
  username = "",
  source = "",
  OnPhotoChange = () => {
    // This is intended
  },
}) => {
  const [previewPhoto, setPreviewPhoto] = useState("");
  const inputImage = useRef(null);

  const handleInputChange = (event) => {
    if (event.target.files && event.target.files[0]) {
      const blobFile = URL.createObjectURL(event.target.files[0]);
      console.log(blobFile);
      setPreviewPhoto(blobFile);
      OnPhotoChange(blobFile);
    }
  };
  return (
    <Stack spacing={1}>
      <Box sx={{ border: "solid 1px #DDDBDA", px: 2, py: 3 }}>
        <Stack direction="row" spacing={4} alignItems="center">
          {source ? (
            <Image
              src={previewPhoto || source}
              width={80}
              height={80}
              style={{ borderRadius: "50%" }}
              alt="photo"
            />
          ) : (
            <Avatar
              {...stringAvatar(username)}
              sx={{ width: 80, height: 80 }}
            ></Avatar>
          )}

          <Button
            onClick={() => {
              inputImage.current.click();
            }}
            sx={{
              border: "solid 1px #6B7789",
              p: 1,
              color: colors.foundationBlue,
              height: 40,
            }}
          >
            <Stack direction="row" alignItems="center" spacing={1}>
              <FileUploadOutlinedIcon sx={{ width: 20, height: 20 }} />
              <span style={{ fontSize: 16, textTransform: "none" }}>
                {source ? "change" : "upload"} photo
              </span>
              <input
                ref={inputImage}
                type="file"
                accept="image/png, image/gif, image/jpeg"
                hidden
                onChange={handleInputChange}
              />
            </Stack>
          </Button>
        </Stack>
      </Box>
      <Typography
        variant="bodySmallMedium"
        sx={{ textAlign: "right", fontStyle: "italic" }}
      >
        *JPG or PNG file formats only. (File size limit is 4 MB)
      </Typography>
    </Stack>
  );
};
