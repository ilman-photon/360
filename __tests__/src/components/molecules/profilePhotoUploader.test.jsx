import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { ProfilePhotoUploader } from "../../../../src/components/molecules/ProfilePhotoUploader/profilePhotoUploader";
import { faker } from "@faker-js/faker";

describe("ProfilePhotoUploader Components", () => {
  let container;
  const landscapeImage = faker.image.imageUrl(275, 173);
  const profilePhoto = {
    name: "my-photo.jpg",
    source: landscapeImage,
  };
  const mockCallBack = jest.fn();
  beforeEach(() => {
    container = render(
      <ProfilePhotoUploader
        source={profilePhoto}
        OnPhotoChange={mockCallBack}
      />
    );
  });

  it("ProfilePhotoUploader render", () => {
    expect(container).toMatchSnapshot();
    expect(
      container.getAllByRole("button", { name: "Change photo" })[0]
    ).toBeVisible();
  });
});
