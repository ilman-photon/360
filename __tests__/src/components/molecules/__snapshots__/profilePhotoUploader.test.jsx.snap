// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePhotoUploader Components ProfilePhotoUploader render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1okj3ks-MuiStack-root"
      >
        <div
          class="MuiBox-root css-vwycly"
        >
          <div
            class="css-uov3go-MuiStack-root"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="photo"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="border-radius: 50%; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1fygijp-MuiTypography-root"
              >
                my-photo.jpg
              </p>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1s7ul64-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="css-1q36rmd-MuiStack-root"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.5608 11.5625H2.96896C2.88303 11.5625 2.81271 11.6328 2.81271 11.7188V12.8906C2.81271 12.9766 2.88303 13.0469 2.96896 13.0469H14.7893L11.9709 16.6211C11.8908 16.7227 11.9631 16.875 12.094 16.875H13.51C13.6057 16.875 13.6955 16.832 13.7561 16.7559L17.0529 12.5742C17.3752 12.1641 17.0842 11.5625 16.5608 11.5625ZM17.0315 6.95312H5.21115L8.02951 3.37891C8.10959 3.27734 8.03732 3.125 7.90646 3.125H6.49045C6.39474 3.125 6.3049 3.16797 6.24435 3.24414L2.94748 7.42578C2.62521 7.83594 2.91623 8.4375 3.43771 8.4375H17.0315C17.1174 8.4375 17.1877 8.36719 17.1877 8.28125V7.10938C17.1877 7.02344 17.1174 6.95312 17.0315 6.95312Z"
                      fill="black"
                    />
                  </svg>
                  <span
                    style="font-size: 16px; text-transform: none;"
                  >
                    Change
                     photo
                  </span>
                  <input
                    accept="image/png, image/gif, image/jpeg"
                    data-testid="loc_uploadProfileImage"
                    hidden=""
                    type="file"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <span
          class="MuiTypography-root MuiTypography-bodySmallMedium css-1ldhyvu-MuiTypography-root"
        >
          *JPG or PNG file formats only. (File size limit is 4 MB)
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1okj3ks-MuiStack-root"
    >
      <div
        class="MuiBox-root css-vwycly"
      >
        <div
          class="css-uov3go-MuiStack-root"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="photo"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="border-radius: 50%; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="css-1okj3ks-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1fygijp-MuiTypography-root"
            >
              my-photo.jpg
            </p>
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1s7ul64-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <div
                class="css-1q36rmd-MuiStack-root"
              >
                <svg
                  fill="none"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.5608 11.5625H2.96896C2.88303 11.5625 2.81271 11.6328 2.81271 11.7188V12.8906C2.81271 12.9766 2.88303 13.0469 2.96896 13.0469H14.7893L11.9709 16.6211C11.8908 16.7227 11.9631 16.875 12.094 16.875H13.51C13.6057 16.875 13.6955 16.832 13.7561 16.7559L17.0529 12.5742C17.3752 12.1641 17.0842 11.5625 16.5608 11.5625ZM17.0315 6.95312H5.21115L8.02951 3.37891C8.10959 3.27734 8.03732 3.125 7.90646 3.125H6.49045C6.39474 3.125 6.3049 3.16797 6.24435 3.24414L2.94748 7.42578C2.62521 7.83594 2.91623 8.4375 3.43771 8.4375H17.0315C17.1174 8.4375 17.1877 8.36719 17.1877 8.28125V7.10938C17.1877 7.02344 17.1174 6.95312 17.0315 6.95312Z"
                    fill="black"
                  />
                </svg>
                <span
                  style="font-size: 16px; text-transform: none;"
                >
                  Change
                   photo
                </span>
                <input
                  accept="image/png, image/gif, image/jpeg"
                  data-testid="loc_uploadProfileImage"
                  hidden=""
                  type="file"
                />
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-bodySmallMedium css-1ldhyvu-MuiTypography-root"
      >
        *JPG or PNG file formats only. (File size limit is 4 MB)
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
